plugins/org.wso2.developerstudio.eclipse.gmf.esb.edit/src/org/wso2/developerstudio/eclipse/gmf/esb/provider/InboundEndpointItemProvider.java

```diff
@@ -4506,47 +4523,59 @@ protected void collectNewChildDescriptors(Collection<Object> newChildDescriptors
      */
     private void updateJavaNamingFactoryInitialProperty(final InboundEndpoint inboundEndpoint, final String newValue) {
 
-        TransactionalEditingDomain domain1 = TransactionUtil.getEditingDomain(inboundEndpoint);
-        domain1.getCommandStack().execute(new RecordingCommand(domain1) {
-            @Override
-            protected void doExecute() {
-                if (!inboundEndpoint.getJavaNamingFactoryInitial().equals(newValue)) {
-                    inboundEndpoint.setJavaNamingFactoryInitial(newValue);
+        TransactionalEditingDomain editingDomain = TransactionUtil.getEditingDomain(inboundEndpoint);
+        try {
+            editingDomain.getCommandStack().execute(new RecordingCommand(editingDomain) {
+                @Override
+                protected void doExecute() {
+                    if (!inboundEndpoint.getJavaNamingFactoryInitial().equals(newValue)) {
+                        inboundEndpoint.setJavaNamingFactoryInitial(newValue);
+                    }
                 }
-            }
-        });
+            });
+        } catch (IllegalStateException e) {
+            log.error("Cannot modify property " + EsbPackage.Literals.INBOUND_ENDPOINT__JAVA_NAMING_FACTORY_INITIAL);
```

Comment:
We need to log the exception as well.

Reply:
We intentionally avoided logging the exception as it dramatically slowed down the loading time

Reply:
When do we get this exception?

Reply:
Only if the user double clicks on the model after selecting a pre-configured profile. However, it will not break the user flow. We had to catch the error to suppress unnecessary popup dialogs shown from the eclipse core.